protos/flow.proto

```diff
@@ -195,9 +195,11 @@ message CreateRawTableOutput { string table_identifier = 1; }
 
 message TableSchema {
   string table_identifier = 1;
+  // DEPRECATED: eliminate when breaking changes are allowed.
   map<string, string> columns = 2;
   repeated string primary_key_columns = 3;
   bool is_replica_identity_full = 4;
+  repeated string column_name_type = 5;
```

Comment:
create a new message `TypedColumn` with two fields, name and type. That will be more verbose but makes the code more readable IMO.

Reply:
Also won't those be boxed per pair? Would rather have two string arrays than that

Reply:
two arrays sound good to me, but FYI, this config is passed around and isn't in the performance critical parts of the code.

Reply:
Updated with two arrays